### ISSUE TITLE
Ignored folders starting with . aren't ignored

### ISSUE THREAD

**Issue by phillip-haydon** [zaggino/brackets-git#1038]
`.gitignore` is setup to ignore anything with say `*.nuget` in it.

Theses are ignored by git, and show as ignored in editors such as Atom. 

In your extension. Nothing that starts with `.` is ever shown as ignored.

If I add a character to the front like `a`. So I have `a.nuget` and update the `.gitignore` it works fine.

**Comment by phillip-haydon** [comment]
Oh wow it's really bad, it doesn't understand .gitignore at all...

```
[Bb]in/
[Oo]bj/
```

It still shows these folders regardless. 


Trying to figure out where you apply the logic for ignores.

**Comment by zaggino** [comment]
`.gitignore` logic is written here: https://github.com/zaggino/brackets-git/blob/master/src/ProjectTreeMarks.js

yes, it doesn't implement complete .gitignore possibilities, feel free to PR it

cc @MarcelGerber

**Comment by phillip-haydon** [comment]
Looking into it now, bit busy this evening tho so will try figure it out this week to see if I can send a PR.

**Comment by phillip-haydon** [comment]
I've fixed it locally:

In `loadIgnoreContents`

```
regex = regex.replace(/\*\*$/g, "(.{0,})").replace(/(\*\*|\*$)/g, "(.+)").replace(/\*/g, "([^/]+)");
```

becomes

```
regex = regex.replace(/\*\*$/g, "(.{0,})").replace(/(\*\*|\*$)/g, "(.+)").replace(/\*/g, "([^/]?)");
```

The last piece instead of `+` (1 or more) I made `*` (0 or more)

This fixes the issue where the folder can be found at any depth in the list, the existing one expected to find it at 1 deeper than the current location when it was 0 deeper than the current location.

And `regexEscape`

```
return str.replace(/([.?+\^$\[\]\\(){}|\-])/g, "\\$1");
```

becomes

```
return str.replace(/([\.\?\+\^$\\(){}|\-])/g, "\\$1");
```

Removing `\[\]` from the list.

This makes sure that [Bb] is treated as a standard regex list of values (in this case looking for case sensitive parsing of the folder where it looks for `Bin` or `bin`)

I've only tested it on a few .NET applications and its now working 100% perfectly for my scenario. Is there anything specific you want me to test? If not I'll send a PR.
